go-controller/pkg/ovn/controller/egress_services/egress_services_controller.go

```diff
@@ -141,33 +141,45 @@ func NewController(
 		workqueue.NewItemFastSlowRateLimiter(1*time.Second, 5*time.Second, 5),
 		"egressservices",
 	)
-	serviceInformer.Informer().AddEventHandler(factory.WithUpdateHandlingForObjReplace(cache.ResourceEventHandlerFuncs{
+	_, err := serviceInformer.Informer().AddEventHandler(factory.WithUpdateHandlingForObjReplace(cache.ResourceEventHandlerFuncs{
 		AddFunc:    c.onServiceAdd,
 		UpdateFunc: c.onServiceUpdate,
 		DeleteFunc: c.onServiceDelete,
 	}))
+	if err != nil {
+		klog.Errorf("Error adding event hander for service informer during egress service controller create : %v", err)
+		return nil, err
```

Comment:
nit: dont need to log the error if you are returning it. The caller should log or propagate it further.